networkx/algorithms/tree/operations.py

```diff
@@ -31,6 +31,8 @@ def join(rooted_trees, label_attribute=None):
     return join_trees(rooted_trees, label_attribute=label_attribute)
 
 
+# Argument types don't match dispatching, but allow manual selection of backend
+@nx._dispatch(graphs=None)
 def join_trees(rooted_trees, *, label_attribute=None, first_label=0):
```

Comment:
FYI, I added this. The input types don't work well with our current dispatching machinery, but this allows `backend=whatever` to be used explicitly.